### PR TITLE
ROL-82: Separate configuration and state - initial impl

### PR DESCRIPTION
**Pull request by g-guirado**
This is not 100% finished, more tests are needed.
However, I think all the functionalities are there and tests are all passing.

Since this is my very first time writing Go code, i'd appreciate if you have any comment before I move even further. Thx!

### CODE REVIEW

**Comment by g-guirado** [comment]
Alright, issue reported by the server is fixed.
Tests are added.
In my opinion, all is there but, again, this is my first PR ever in Go.

File: core/network/configuration_fetcher.go [hunk 1]
```diff
@@ -40,10 +42,19 @@ func (f *configurationFetcher) Fetch() *configuration.FetchResult {
 	}
 
 	if fetchResult.IsSuccessStatusCode() {
-		return configuration.NewFetchResult(string(fetchResult.Content), source)
+		configurationFetchResult := configuration.NewFetchResult(string(fetchResult.Content), source)
+		if configurationFetchResult == nil {
+			return nil
+		}
+
+		if configurationFetchResult.ParsedData.Result == 404 {
```
Comment (AsafRollout) [inline core/network/configuration_fetcher.go]: what happens here if ParsedData.Result doesn't exits (regular configuration)?

Comment (AsafRollout) [reply core/network/configuration_fetcher.go]: my bad

**Comment by g-guirado** [comment]
Little proof video that all works fine: https://drive.google.com/open?id=1bQSaapR78MG0jSGurAsqQkiWCZCkOK6W

**Pull request merged**